Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -60,55 +62,73 @@ private void AddDirectiveMiddlewareAndCommonCommandHandlers()
-                            context,
-                            next),
-
-                        LoadExtension loadExtension =>
-                        HandleLoadExtension(
-                            loadExtension,
-                            context, 
-                            next),
-
-                        DisplayValue displayValue =>
-                        HandleDisplayValue(
-                            displayValue,
-                            context,
-                            next),
-
-                        UpdateDisplayedValue updateDisplayValue =>
-                        HandleUpdateDisplayValue(
-                            updateDisplayValue,
-                            context,
-                            next),
+                            SubmitCode submitCode =>
+                            HandleDirectivesAndSubmitCode(
+                                submitCode,
+                                context,
+                                next),
+
+                            LoadExtension loadExtension =>
+                            HandleLoadExtension(
+                                loadExtension,
+                                context,
+                                next),
+
+                            DisplayValue displayValue =>
+                            HandleDisplayValue(
+                                displayValue,
+                                context,
+                                next),
+
+                            UpdateDisplayedValue updateDisplayValue =>
+                            HandleUpdateDisplayValue(
+                                updateDisplayValue,
+                                context,
+                                next),
+
+                            LoadExtensionFromNuGetPackage loadCSharpExtension =>
+                            HandleLoadExtensionInNuGetPackage(
+                                loadCSharpExtension,
+                                context,
+                                next),
 
                             _ => next(command, context)
                         });
         }
 
+        private async Task HandleLoadExtensionInNuGetPackage(
+            LoadExtensionFromNuGetPackage loadExtensionFromNuGetPackage,
+            KernelInvocationContext invocationContext,
+            KernelPipelineContinuation next)
+        {
+            loadExtensionFromNuGetPackage.Handler = async context =>
+            {
+                if (invocationContext.HandlingKernel is IExtensibleKernel extensibleKernel)
+                {
+                    if (NuGetPackagePathResolver.TryGetNuGetPackageBasePath(loadExtensionFromNuGetPackage.NugetPackageReference, loadExtensionFromNuGetPackage.MetadataReferences, out var nugetPackageDirectory))
+                    {
+                        await extensibleKernel.LoadExtensionsInDirectory(nugetPackageDirectory, context);
+                    }
+                }
+
+                context.Complete();
```

Comment:
You shouldn't call complete if you plan to call the continuation as this will prevent later steps in the pipeline from publishing events. `C

Reply:
Yes, I removed the call to next since it is the only place that command needs to be handled.